src/core/Proc/Async.pm6

```diff
@@ -271,17 +271,21 @@ my class Proc::Async {
         $promise;
     }
 
+    has $!start-lock = Lock.new;
     method start(Proc::Async:D: :$scheduler = $*SCHEDULER, :$ENV, :$cwd = $*CWD) {
         X::Proc::Async::AlreadyStarted.new(proc => self).throw if $!started;
-        $!started = True;
+        $!start-lock.protect: {
```

Comment:
Does all of this logic need to be inside the lock, or could would the following suffice?
```
method start(Proc::Async:D: :$scheduler = $*SCHEDULER, :$ENV, :$cwd = $*CWD) {
    $!start-lock.protect: {
        X::Proc::Async::AlreadyStarted.new(proc => self).throw if $!started;
        $!started = True;
    }

    ...
}
```

Reply:
No need for a lock at all, just use `cas` on the `$!started`.

Reply:
I updated the branch to use ```cas``` as suggested